plugin.py

```diff
@@ -0,0 +1,136 @@
+        session = self.weaksession()
+        if session:
+            session.window.show_input_panel("Open this url in the browser and paste in its place the redirection:", link,
+                                     lambda r: self._send_response(r), None, lambda: self._send_response(None))
+        
+    def _send_response(self, input: Optional[str]) -> None:
+        if input:
+            res = urlparse(input)
+            assert(res.scheme == 'vscode')
+            request = Request(self.cmd_callback, input, None, progress=True)
+            session = self.weaksession()
+            if session:
+                session.send_request(request, lambda p: print("Connected." + str(p)),
+                    lambda r: print("Invalid authorization-response url"))
+        else:
+            print("Logging cancelled")
+
+
+class LspGrammarlyExecuteLogoutCommand(LspGrammarlyCommand):
+    cmd = "$/logout"
+
+    def run(self, edit: sublime.Edit) -> None:
+        def run_async() -> None:
+            session = self.session_by_name()
+            assert(session)
+            request = Request(self.cmd, None, None, progress=True)
+            session.send_request(request, self._handle_response,
+                lambda r: print("Failed to logout: " + str(r)))
+        sublime.set_timeout_async(run_async)
+
+    def _handle_response(self, response: Union[List[Location], None]) -> None:
+        if response:
+            print("Logout response: " + str(response))
+        else:
+            print("Logged out")
+
+class LspGrammarlyExecuteIsConnectedCommand(LspGrammarlyCommand):
+    cmd = "$/isUserAccountConnected"
+
+    def run(self, edit: sublime.Edit) -> None:
+        def run_async() -> None:
+            session = self.session_by_name()
+            assert(session)
+            request = Request(self.cmd, None, None, progress=True)
+            session.send_request(request, self._handle_response,
+                lambda r: print("Failed to query whether connected: " + str(r)))
+        sublime.set_timeout_async(run_async)
+
+    def _handle_response(self, response: Union[List[Location], None]) -> None:
+        if response:
+            print("Connected: " + str(response))
+        else:
+            print("Not connected")
+
+class LspGrammarlyPlugin(NpmClientHandler):
+    package_name = "LSP-Grammarly"
+    server_directory = 'grammarly-languageserver'
+    server_binary_path = os.path.join(
+        server_directory, 'node_modules', 'grammarly-languageserver', 'bin', 'server.js'
+    )
+    
+    @classmethod
+    def name(cls) -> str:
+        return LspGrammarlyCommand.session_name
+    
+    @classmethod
+    def minimum_node_version(cls) -> Tuple[int, int, int]:
+        return (16, 13, 0)
+    
+    def handle_status_response(self, status, is_error):
+        if is_error:
+            print("Requesting status failed")
+    
+    def m___onDocumentStatus(self, params):
+        print(params)
```

Comment:
Do we have to log this? If yes, then please make it clear that it's coming from this package (by prefixing it with `LSP-Grammarly` for example). If not then we can just use `pass` instead.

Reply:
I moved it to the status bar. It may take a few seconds for the server to come back with the new suggestions, so it's nice to have a status update (instead of completely removing it).